python/pyspark/sql/pandas/serializers.py

```diff
@@ -226,9 +225,6 @@ def create_array(s, t):
                 s = _check_series_convert_timestamps_internal(s, self._timezone)
             elif t is not None and pa.types.is_map(t):
                 s = _convert_dict_to_map_items(s)
-            elif is_categorical_dtype(s.dtype):
-                # Note: This can be removed once minimum pyarrow version is >= 0.16.1
-                s = s.astype(s.dtypes.categories.dtype)
```

Comment:
@BryanCutler Seems like if `t is None`, `pa.Array.from_pandas(s, mask=mask, type=t, safe=self._safecheck)` handles the categorical type as integer (`tinyint` as its code) instead of some type of `s.dtypes.categories.dtype`. Is that an expected behavior?

Reply:
ah, it will be `dictionary<values=string, indices=int8, ordered=0>` if we don't specify `type`.

Reply:
so we can't remove it if `t is None`. or we need to support `dictionary` type.